src/storage/v3/shard_rsm.cpp

```diff
@@ -263,28 +275,25 @@ msgs::ReadResponses ShardRsm::HandleRead(msgs::ScanVerticesRequest &&req) {
   for (auto it = vertex_iterable.begin(); it != vertex_iterable.end(); ++it) {
     const auto &vertex = *it;
 
-    if (start_ids == vertex.PrimaryKey(View(req.storage_view)).GetValue()) {
+    if (start_ids <= vertex.PrimaryKey(View(req.storage_view)).GetValue()) {
```

Comment:
Giving up on sending the optional since it introduced additional complexity:
-  Handling `nullopt` in `Request method 
- Handling `nullopt` in `InitializationOfExecutionState`
 plus this solution works and can be used in the same way, the client can always send global min primary key value to get everything